### ISSUE TITLE
Error downloading and uploading (0.4.0)

### ISSUE THREAD

**Issue by aptonline** [jhalter/mobius#24]
Making a few tests with the latest build (0.4.0) using Nostalgia on Mac OS X (not sure if this client is supported) and receiving the following error when attempting to upload a small test file (TXT 27 bytes).

![Screenshot 2022-06-05 at 10 30 06](https://user-images.githubusercontent.com/190136/172044466-13b5b38c-fa6e-4785-a6bc-6d4562bc133d.png)
The uploaded file attempt does show as .incomplete in the files directory:

![Screenshot 2022-06-05 at 10 30 42](https://user-images.githubusercontent.com/190136/172044469-07c561ea-98aa-4c9b-8452-5b9122057000.png)

And this is the log which crashed the server:

```
info	File upload started	{"transactionRef": "4sZJgQ==", "dstFile": "/usr/local/var/mobius/config/Files///test_upload.txt"}
panic: runtime error: slice bounds out of range [:89] with capacity 87

goroutine 40 [running]:
github.com/jhalter/mobius/hotline.(*FlatFileInformationFork).UnmarshalBinary(0xc000096000?, {0xc00005a0c0?, 0x71b320?, 0x8c4110?})
	/app/hotline/flattened_file_object.go:119 +0x3e7
github.com/jhalter/mobius/hotline.receiveFile({0x719940, 0xc000096000}, {0x719a00?, 0xc00000e098?}, {0x71a140?, 0x8c3ea0?})
	/app/hotline/transfer.go:63 +0x2c5
github.com/jhalter/mobius/hotline.(*Server).handleFileTransfer(0xc0001a0000, {0x7f7a144a4b58?, 0xc000096000})
	/app/hotline/server.go:739 +0x45c
github.com/jhalter/mobius/hotline.(*Server).ServeFileTransfers.func1()
	/app/hotline/server.go:95 +0x5a
created by github.com/jhalter/mobius/hotline.(*Server).ServeFileTransfers
	/app/hotline/server.go:94 +0x11e
```

I also attempted to download the small `hello.txt` file and received the following.

![Screenshot 2022-06-05 at 10 43 17](https://user-images.githubusercontent.com/190136/172044659-f557bea6-11b3-4039-8288-5b09c6ce0bd8.png)
But no error in the log.

`info	File download started	{"filePath": "/usr/local/var/mobius/config/Files/hello.txt", "transactionRef": "aB0Nhg=="}
`

**Comment by jhalter** [comment]
Thanks for the detailed info.  I haven't tested with Nostalgia, but I'll see if I can reproduce this.

**Comment by aptonline** [comment]
Its no big deal if not, I now have sheep shaver setup on my laptop so connecting via Hotline Client 1.2.3 under Mac OS 7.6.

**Comment by jhalter** [comment]
Ok, I was able to reproduce it so that helps a bunch.  I'll take a look.

**Comment by jhalter** [comment]
This specific issue should be fixed in v0.5.0, but I'm sure there are more compatibility problems to uncover.

**Comment by jhalter** [comment]
This should be fix, but will revisit Nostalgia compatibility in #26
